src/core/Akka.Remote/Serialization/DaemonMsgCreateSerializer.cs

```diff
@@ -114,6 +114,17 @@ public override object FromBinary(byte[] bytes, Type type)
         {
             var proto = DaemonMsgCreateData.ParseFrom(bytes);
             var clazz = Type.GetType(proto.Props.Clazz);
+
+            if (clazz == null)
+            {
+                var msg =
+                    string.Format(
+                        "Could not find '{0}' in loaded assemblies. If this is a remote/cluster deploy scenario, " +
+                        "ensure the remote system has referenced the assembly containing this type.", proto.Props.Clazz);
+                    
+                throw new ActorInitializationException(msg);
```

Comment:
Good point - the message from `TypeLoadException` is as follows:
```
TypeLoadException: Could not load type 'TypeName' from assembly 
'AssemblyName, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
```
For someone familiar with the exception, it will be clear that the remote system does not have access either to the containing assembly or one of its dependencies. For someone unfamilar, I still think it would be enough to get started with. So I am in favour and can make the change if you're happy with that. 

I guess the case for wrapping it would be that it could make it immediately obvious to the user that a known specific issue with a known specific solution (review the referenced/available assemblies) had occurred. But going down that track can get you into trouble if later on another set of circumstances can trigger the same exception but with a different cause/resolution - now the error message is too specific to the old problem and inaccurate for this new one.